config/routes.rb

```diff
@@ -1,6 +1,32 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  devise_for :users, skip: [:sessions]
+  
+  as :user do
+    get 'login', to: 'devise/sessions#new', as: :new_user_session
+    post 'login', to: 'devise/sessions#create', as: :user_session
+    delete 'logout', to: 'devise/sessions#destroy', as: :destroy_user_session
+  end
+  
+  root 'users#index'
+  
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :likes, only: [:create, :destroy]
+      resource :deletion, only: [:destroy], controller: 'post_deletions'
+      resources :comments, only: [:new, :create, :show] do
+        resource :deletion, only: [:destroy], controller: 'comment_deletions'
+      end
+      
+    end
+  end
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  namespace :api do
+    namespace :v1 do
+      resources :users, only: [] do
+        resources :posts, only: [:index] do
+          resources :comments, only: [:index, :create]
+        end
+      end
+    end
```

Comment:
- [ ] As required for the project all API your endpoints to receive JSON and responds in a JSON format as well. Right now, anyone of them is responding as expected. Kindly ensure that you fix them. I can see all the endpoints but not working.
![image](https://github.com/EvansSnave/Blog_app_rails/assets/83042742/4f7af77c-8544-4504-9d8e-6efa0fc3df92)